### PR TITLE
Dev/issue 11075 perms

### PR DESCRIPTION
**Pull request by mcantelon**
* Added group-based security restrictions to a number of actions that didn’t have any
* Replaced some security-related action logic with security.yml rules
* Added method to user class for checking, using security.yml files, if a user has permission to use a module’s action
* Added logic to the menu model to check if a user has permission to access an action before displaying a menu item for it
* Added logic to limit display of the add menu to groups that can add content

### CODE REVIEW

File: lib/myUser.class.php [hunk 1]
```diff
@@ -207,6 +211,66 @@ public function listGroups()
...
+    if ($file = sfContext::getInstance()->getConfigCache()->checkConfig($securityFilePath, true))
+    {
+      require($file);
+    }
+
+    // Get credentials, using security.yml parsing convention
+    $credentials = $this->getModuleSecurityValue($action, 'credentials');
+
+    // Allow access if action isn't secured or user has appropriate credentials
+    if (!$this->getModuleSecurityValue($action, 'is_secure', false) || $this->hasCredential($credentials))
```
Comment (jraddaoui) [inline lib/myUser.class.php]: We could return the condition directly.

File: lib/myUser.class.php [hunk 2]
```diff
@@ -207,6 +211,66 @@ public function listGroups()
...
+   *
+   * @return boolean
+   */
+  public function checkModuleActionAccess($module, $action)
+  {
+    // Set security property to module's security configuration
+    $securityFilePath = 'modules/'. $module .'/config/security.yml';
+    if ($file = sfContext::getInstance()->getConfigCache()->checkConfig($securityFilePath, true))
+    {
+      require($file);
```
Comment (jraddaoui) [inline lib/myUser.class.php:235]: Is this loading the config. file inside `$this->security`? It'd be a lot clearer with `sfYaml::load()`.

**Review by jraddaoui — CHANGES_REQUESTED** [review]
Looks great! Just a couple of minor suggestions.

File: lib/myUser.class.php (hunk 2 above)
Comment (mcantelon) [reply lib/myUser.class.php:235]: It's using a mechanism, used in other permission processing in Symfony, that returns the filename of a PHP file if it has already been parsed from YAML, so it should be slightly more performant.

I could always add logic to it does something similar with another layer of caching though.

Comment (jraddaoui) [reply lib/myUser.class.php:235]: I didn't know, thanks @mcantelon!

Could you add a comment in the code explaining that? I think it will be helpful.

Comment (mcantelon) [reply lib/myUser.class.php:235]: Will do... thanks!

**Pull request merged**